hugolib/page__meta.go

```diff
@@ -167,6 +167,10 @@ func (p *pageMeta) Description() string {
 	return p.description
 }
 
+func (p *pageMeta) Environment() string {
```

Comment:
0K, this PR looks mostly good, but we don't want to add the `Environment()` everywhere -- the Page interface is big as it is.

So I suggest that you in the page matcher do something ala:

```go
if m.Environment != "" {
		g, err := glob.GetGlob(m.Environment)
		if err == nil && !g.Match(p.Site().Hugo().Environment()) {
			return false
		}
	}

```

Reply:
Yea, I see your point. Ultimately we may want to split it (e.g. having both a MatchesSite and a Matches Page method), but even then you will have the same test challenge. 

There is a `testSite` implementation in there somewhere, so I suggest you just expand on that. It makes for a little more work, but certainly a better end game.

Reply:
I think the most recent change should address the requested change. Let me know what you think!